src/IRrecv.cpp

```diff
@@ -356,23 +369,43 @@ void IRrecv::enableIRIn(const bool pullup) {
     pinMode(params.recvpin, INPUT);
 #endif  // UNIT_TEST
   }
+
 #if defined(ESP32)
   // Initialise the ESP32 timer.
+#if defined(ESP_ARDUINO_VERSION) && \
+    (ESP_ARDUINO_VERSION >= ESP_ARDUINO_VERSION_VAL(3, 0, 0))
+    // Use newer timerBegin signature for ESP32 core version 3.x
+    timer = timerBegin(1000000);  // Initialize with 1MHz (1us per tick)
+    Serial.println("Starting timer initialization...");
```

Comment:
serial.print is a no go!

Reply:
Thanks @Jason2866 It was for debug finish to test and removed this from code